### DISCUSSION TITLE
How to set community?

### DISCUSSION THREAD

**Discussion by sje30** [eblondel/zen4R#162] [Q&A] [answered]
Hi,

thank you for working on the new interface!  I seem to be making much better progress now.

One issue: before I could do something like:

```
draft$addCommunity("codecheck")
```
following your example in the vignette:

```
myrec$addCommunity("ecfunded")
```

but when I do ls(draft) I don't see the addCommunity method; is there another way of adding the community now?

Thanks!

I'm using master from GitHub, but I see the package is now going across to CRAN.  Thanks again for all the work with this big update to the API.

**Comment by sje30** [comment]
I also found some of the other examples in the vignette now do not work:

(sec 4.4.2 of https://cran.r-project.org/web/packages/zen4R/vignettes/zen4R.html#ZenodoDeposit-emptyRecord)

```
myrec$setUploadType("software")
myrec$addCreator(firstname = "John", lastname = "Doe", affiliation = "Independent", orcid = "0000-0000-0000-0000")

```

becomes

```
myrec$setResourceType("software")
myrec$addCreator(firstname = "John", lastname = "Doe", orcid = "0000-0000-0000-0000")
```

**Comment by eblondel** [comment, marked as answer]
Sorry indeed there were some changes in the API regarding the communities, and i forget to update the vignette in this sense.
They are 2 different ways to assign communities:


* for a **published** record:  `ZENODO$submitRecordToCommunities(record, communities)` (with `ZENODO` being the instance of `ZenodoManager` you created). You can do this to multiple communities, but your record should be already published. This will create a `request` for inclusion in the (list of) community.

As submitter of a request, you have privilege to cancel it. As manager of the community (if you submitted a record to a community you manage), you can either accept or decline the request for addition.
You can search for requests with following code, and then applyl cancel/accept/decline:

```R
my_community = "<your community here>"
zen_com = ZENODO$getCommunityById(my_community)
#retrieve the request that was created with the review submission (using status the target community and the target record)
reqs = ZENODO$getRequests(q = sprintf("status:submitted AND receiver.community:%s AND topic.record:%s", zen_com$id, record$id))

#as submitter you can cancel it
ZENODO$cancelRequest(reqs[[1]]$id) #return TRUE if canceled
#as manager you can either accept/decline
ZENODO$acceptRequest(reqs[[1]]$id) #return TRUE if accepted
ZENODO$declineRequest(reqs[[1]]$id) #return TRUE if accepted
```

* for a **draft** record: you can create a request for addition of community, which is the analog of what we had before the upgrade to Invenio. Before, the addition of a community was done at metadata level, now it is done with 2 different methods:

    * a first method that will "attach" the record the community: `ZENODO$createReviewRequest(record, my_community)`
    * a second method that allows to submit the record for review by this community manager: `ZENODO$submitRecordForReview(record$id)`. Indeed adding a draft deposit to a community means now that you are requesting a record to be reviewed for publication in a community. The latter `submitRecordForReview` adds a `review` step before the record publication. This is materialized by a `request` for addition to the community (as above), but with a review to be performed by the manager. This has several implications:
        * you can't anymore publish by yourself the record, this is delegated to the community manager. 
        * the community manager has editing rights over your draft deposit

Note that:
- you can create a review request to only **one** community. As far as I could dig, this is a limitation of Zenodo/Invenio
- in case you want to cancel the review, you have to search for the community requests (see above code), and cancel you request. After this you will have to delete the community attachment by doing `ZENODO$deleteReviewRequest(record)` Regarding this you can see this discussion: https://github.com/eblondel/zen4R/discussions/158

Hope this helps

**Reply by sje30** [reply]
Thank you very much for this detailed explanation.  I will try it out.

**Reply by eblondel** [reply]
I postponed a bit the publication on CRAN ,to make sure the vignette is updated. I also miss the refactoring on the grants/awards that I forgot to implement.
